app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -442,4 +468,77 @@ public void run() {
             },UNDELIVERED_PACKAGE_TIMEOUT);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * this functions converts a byte array into a file
+     * @param fileName name of the file with extension
+     * @param fileByte file bytes to convert
+     * @return a path to the newly written file
+     */
+    public String writeFileExternalStorage(String fileName, byte[] fileByte) {
+        String state = Environment.getExternalStorageState();
+        //external storage availability check
+        if (!Environment.MEDIA_MOUNTED.equals(state)) {
+            return null;
+        }
+        File file = new File(Environment.getExternalStoragePublicDirectory(
+                Environment.DIRECTORY_DOWNLOADS), fileName);
+        FileOutputStream outputStream = null;
+        try {
+            file.createNewFile();
+            //second argument of FileOutputStream constructor indicates whether to append or create new file if one exists
+            outputStream = new FileOutputStream(file, false);
+            outputStream.write(fileByte);
+            outputStream.flush();
+            outputStream.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return file.getPath();
+    }
+
+    /**
+     * this function turns any file into byte arrays
+     * @param file the file to turn byte arrays into
+     * @return  byte arrays
+     * @throws IOException
+     */
+    public static byte[] getBytesFromFile(File file) throws IOException {
+        InputStream fileInputStream = new FileInputStream(file);
+
+        // Get the size of the file
+        long length = file.length();
+
+        // Before converting to an int type, check
+        // to ensure that file is not larger than Integer.MAX_VALUE.
+        if (length > Integer.MAX_VALUE) {
+            // File is too large
```

Comment:
What needs to happen here? Why do we check this?

Reply:
Ah forgot to throw an exception. but basically length will determine file size in bytes. I am not sure if we should limit this to something like 100 mb.